### PR TITLE
Make routes by json config

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**

### CODE REVIEW

File: src/config/appConfig.json [hunk 1]
```diff
@@ -177,5 +177,67 @@
       "label": "Last Modified Date",
       "key": "lastModifiedDate"
     }
+  ],
+  "RouteArray": [
+    {
+      "routName": "StudentPage",
```
Comment (pepcus-pratik) [inline src/config/appConfig.json]: Why you have two different spelling of word `route` in the code. Fix this typo throughout the code base.

File: src/config/appConfig.json [hunk 2]
```diff
@@ -177,5 +177,67 @@
       "label": "Last Modified Date",
       "key": "lastModifiedDate"
     }
+  ],
+  "RouteArray": [
```
Comment (pepcus-pratik) [inline src/config/appConfig.json]:
```suggestion
  "routes": [
```

File: src/config/appConfig.json [hunk 3]
```diff
@@ -177,5 +177,67 @@
...
+    {
+      "routName": "StudentPage",
+      "routePath": "/",
+      "componentName": "StudentPage",
```
Comment (pepcus-pratik) [inline src/config/appConfig.json]:
```suggestion
      "component": "StudentPage",
```

File: src/config/appConfig.json [hunk 4]
```diff
@@ -177,5 +177,67 @@
       "label": "Last Modified Date",
       "key": "lastModifiedDate"
     }
+  ],
+  "RouteArray": [
+    {
+      "routName": "StudentPage",
+      "routePath": "/",
```
Comment (pepcus-pratik) [inline src/config/appConfig.json]:
```suggestion
      "path": "/",
```

File: src/config/appConfig.json (hunk 1 above)
Comment (pepcus-pratik) [inline src/config/appConfig.json]:
```suggestion
      "routName": "default",
```

File: src/config/appConfig.json [hunk 5]
```diff
@@ -177,5 +177,67 @@
...
+      "isActive": true
+    },
+    {
+      "routName": "admin",
+      "routePath": "/admin",
+      "componentName": "SplashPage",
+      "isActive": true
+    },
+    {
+      "routName": "student-search",
```
Comment (pepcus-pratik) [inline src/config/appConfig.json]:
```suggestion
      "routName": "memberSearch",
```

File: src/config/appConfig.json [hunk 6]
```diff
@@ -177,5 +177,67 @@
...
+      "isActive": true
+    },
+    {
+      "routName": "home",
+      "routePath": "/home",
+      "componentName": "SplashPagePrePopulated",
+      "isActive": false
+    },
+    {
+      "routName": "studentRegister",
```
Comment (pepcus-pratik) [inline src/config/appConfig.json]:
```suggestion
      "routName": "memberRegister",
```

File: src/config/appConfig.json [hunk 7]
```diff
@@ -177,5 +177,67 @@
...
+      "routePath": "/studentRegister",
+      "componentName": "StudentRegistrationForm",
+      "isActive": true
+    },
+    {
+      "routName": "studentCorrection",
```
Comment (pepcus-pratik) [inline src/config/appConfig.json]:
```suggestion
      "routName": "registrationCorrection",
```

File: src/config/appConfig.json [hunk 8]
```diff
@@ -177,5 +177,67 @@
...
+      "isActive": true
+    },
+    {
+      "routName": "adminPanel",
+      "routePath": "/adminPanel",
+      "componentName": "AdminPanel",
+      "isActive": false
+    },
+    {
+      "routName": "student-login",
```
Comment (pepcus-pratik) [inline src/config/appConfig.json]:
```suggestion
      "routName": "memberLogin",
```

File: src/components/coreComponents/Routes.jsx [hunk 9]
```diff
@@ -37,7 +31,23 @@ class Routes extends Component {
       });
     }
   }
-
+  getRouts = Consumer => RouteArray.map((routeObject) => {
+    const ComponentName = components[routeObject.componentName];
+    if (routeObject.isActive) {
+      return (
+        <Route
+          key={uniqueId(routeObject.componentName)}
```
Comment (pepcus-pratik) [inline src/components/coreComponents/Routes.jsx]: `uniqueId` not needed here. Use the route's name.

File: src/components/coreComponents/Routes.jsx [hunk 10]
```diff
@@ -1,21 +1,15 @@
...
 import Loader from '../Loader';
-import SplashPagePrePopulated from '../SplashPagePrePopulated';
-import AdminPanel from '../AdminPanel';
-import StudentInformationGrid from '../StudentInformationGrid';
-import StudentCredentialPage from '../StudentCredentialPage';
-import StudentPage from '../StudentPage';
 import Context from './ConfigProvider';
-import ParentsRegistration from '../ParentsRegistrationForm';
-import Files from '../Files';
+import components from '../routeComponents/index';
```
Comment (pepcus-pratik) [inline src/components/coreComponents/Routes.jsx]:
```suggestion
import RouteComponents from '../routeComponents';
```

File: src/config/appConfig.json (hunk 1 above)
Comment (pepcus-pratik) [inline src/config/appConfig.json]:
```suggestion
      "name": "StudentPage",
```

File: src/components/coreComponents/Routes.jsx [hunk 11]
```diff
@@ -37,7 +31,23 @@ class Routes extends Component {
       });
     }
   }
-
+  getRouts = Consumer => RouteArray.map((routeObject) => {
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/coreComponents/Routes.jsx]: Done.

File: src/components/coreComponents/Routes.jsx [hunk 12]
```diff
@@ -37,7 +31,23 @@ class Routes extends Component {
...
+    const ComponentName = components[routeObject.componentName];
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/coreComponents/Routes.jsx]: Done.

File: src/components/coreComponents/Routes.jsx [hunk 13]
```diff
@@ -46,96 +56,7 @@ class Routes extends Component {
...
-          <Route
-            exact
-            path="/files"
-            component={() => (
-              <Consumer>
-                {context => <Files context={context} />}
-              </Consumer>
-            )}
-          />
+          {this.getRouts(Consumer)}
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/coreComponents/Routes.jsx]: Done.

File: src/config/appConfig.json (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/config/appConfig.json]: Done.

File: src/config/appConfig.json (hunk 2 above)
Comment (BhagyashreeBhoyar) [reply src/config/appConfig.json]: Done.

File: src/config/appConfig.json (hunk 3 above)
Comment (BhagyashreeBhoyar) [reply src/config/appConfig.json]: Done.

File: src/config/appConfig.json (hunk 4 above)
Comment (BhagyashreeBhoyar) [reply src/config/appConfig.json]: Done.

File: src/config/appConfig.json (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/config/appConfig.json]: Done.

File: src/config/appConfig.json (hunk 5 above)
Comment (BhagyashreeBhoyar) [reply src/config/appConfig.json]: Done.

File: src/config/appConfig.json (hunk 6 above)
Comment (BhagyashreeBhoyar) [reply src/config/appConfig.json]: Done.

File: src/components/coreComponents/Routes.jsx (hunk 9 above)
Comment (BhagyashreeBhoyar) [reply src/components/coreComponents/Routes.jsx]: Done.

File: src/components/coreComponents/Routes.jsx (hunk 10 above)
Comment (BhagyashreeBhoyar) [reply src/components/coreComponents/Routes.jsx]: Done.

File: src/config/appConfig.json (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/config/appConfig.json]: Done.

File: src/components/routeComponents/index.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
+import SplashPage from '../SplashPage';
+import StudentRegistrationForm from '../StudentRegistrationForm';
+import StudentRegistrationCorrectionForm from '../StudentRegistrationCorrectionForm';
+import SplashPagePrePopulated from '../SplashPagePrePopulated';
+import AdminPanel from '../AdminPanel';
+import StudentInformationGrid from '../StudentInformationGrid';
+import StudentCredentialPage from '../StudentCredentialPage';
+import StudentPage from '../StudentPage';
+import ParentsRegistration from '../ParentsRegistrationForm';
+import Files from '../Files';
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/routeComponents/index.js]: Done.

File: src/config/appConfig.json (hunk 7 above)
Comment (BhagyashreeBhoyar) [reply src/config/appConfig.json]: Done.

File: src/config/appConfig.json (hunk 8 above)
Comment (BhagyashreeBhoyar) [reply src/config/appConfig.json]: Done.

**Pull request merged**